fuse/eval/metrics/metrics_common.py

```diff
@@ -239,15 +243,41 @@ def sync_ids(ids: List[Tuple[str, int]]) -> List[Any]:
         return data_list
 
     @staticmethod
-    def _df_dict_apply(data: pd.Series, func: Callable) -> pd.Series:
-        result = func(NDict(data.to_dict()))
-        return pd.Series(result.flatten())
+    def _df_dict_apply(
+        data: pd.Series, func: Callable, batch: bool = False
+    ) -> pd.Series:
+        if batch:
```

Comment:
Not sure I understood the use-case: `data` is always a single data point (e.g. series) and `func` might be perform on a batch level? And where `func` expects a batch you pass `batch=True`?

I'm suggesting to rename `batch` or add a short description to make it clearer :))

Reply:
The metric has some preprocessing func in a batch level.
Here we are not running a training loop,, to avoid duplication, we run the function on each sample as if it was a batch. To do that we are converting single sample to batch, call the function and then squeezed it back to sample.
As I mentioned, those are dark places in fuse, that I would refactor If we had some time to spend on it.